src/util.cpp

```diff
@@ -889,6 +889,97 @@ QSize CMinimumStackedLayout::sizeHint() const
+        // HostAddr.protocol() ) );
+        if ( HostAddr.protocol() == QAbstractSocket::IPv4Protocol || ( bEnableIPv6 && HostAddr.protocol() == QAbstractSocket::IPv6Protocol ) )
+        {
+            InetAddr = HostAddr;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool NetworkUtil::ParseNetworkAddressSrv ( QString strAddress, CHostAddress& HostAddress, bool bEnableIPv6 )
+{
+    // init requested host address with invalid address first
+    HostAddress = CHostAddress();
+
+    QRegularExpression plainHostRegex ( "^([^\\[:0-9.][^:]*)$" );
+    if ( plainHostRegex.match ( strAddress ).capturedStart() != 0 )
+    {
+        // not a plain hostname? then don't attempt SRV lookup and fail
+        // immediately.
+        return false;
+    }
+
+    QDnsLookup* dns = new QDnsLookup();
+    dns->setType ( QDnsLookup::SRV );
+    dns->setName ( QString ( "_jamulus._udp.%1" ).arg ( strAddress ) );
+    dns->lookup();
+    // QDnsLookup::lookup() works asynchronously. Therefore, wait for
+    // it to complete here by resuming the main loop here.
+    // This is not nice and blocks the UI, but is similar to what
+    // the regular resolve function does as well.
+    QTime dieTime = QTime::currentTime().addMSecs ( DNS_SRV_RESOLVE_TIMEOUT_MS );
+    while ( QTime::currentTime() < dieTime && !dns->isFinished() )
+    {
+        QCoreApplication::processEvents ( QEventLoop::ExcludeUserInputEvents, 100 );
+    }
+    QList<QDnsServiceRecord> records = dns->serviceRecords();
+    dns->deleteLater();
+    if ( records.length() != 1 )
+    {
+        return false;
+    }
+    QDnsServiceRecord record = records.first();
+    if ( record.target() == "." || record.target() == "" )
+    {
+        // RFC2782 says that "." indicates that the service is not available.
+        // (Qt strips the trailing dot, which is why we check for empty string
+        // as well)
+        // Therefore, end processing here (= return true), but pass back an
+        // invalid HostAddress to let the connect logic fail properly.
+        HostAddress = CHostAddress ( QHostAddress ( "." ), 0 );
+        return true;
+    }
+    qDebug() << qUtf8Printable (
+        QString ( "resolved %1 to a single SRV record: %2:%3" ).arg ( strAddress ).arg ( record.target() ).arg ( record.port() ) );
+
+    QHostAddress InetAddr;
+    if ( ParseNetworkAddressName ( record.target(), InetAddr, bEnableIPv6 ) )
+    {
+        HostAddress = CHostAddress ( InetAddr, record.port() );
+        return true;
+    }
+    return false;
+}
+
+bool NetworkUtil::ParseNetworkAddressWithSrvDiscovery ( QString strAddress, CHostAddress& HostAddress, bool bEnableIPv6 )
+{
+    // Try SRV-based discovery first:
+    if ( ParseNetworkAddressSrv ( strAddress, HostAddress, bEnableIPv6 ) )
+    {
+        return true;
+    }
+    // Try regular connect:
+    return ParseNetworkAddress ( strAddress, HostAddress, bEnableIPv6 );
+}
+
```

Comment:
Could you add a new comment to say something like this expects to find the address either from a valid IP numeric value or a name via DNS lookup through A or AAAA records.

Reply:
Oh - note that, given the RFC says the SRV entry can point to an AAAA record, `ParseNetworkAddress` might not quite work.  I think it expects IPv6 addresses to be numeric? `QRegularExpression rx1 ( "^\\[([^]]*)\\](?::(\\d+))?$" );`

Reply:
> Could you add a new comment to say something like this expects to find the address either from a valid IP numeric value or a name via DNS lookup through A or AAAA records.

Added.

> Oh - note that, given the RFC says the SRV entry can point to an AAAA record, `ParseNetworkAddress` might not quite work. I think it expects IPv6 addresses to be numeric? `QRegularExpression rx1 ( "^\\[([^]]*)\\](?::(\\d+))?$" );`

Not sure I understand. I think plain IPs in SRV records aren't supposed to be supported. SRV records return names which can be looked up to A or AAAA records. Shouldn't ParseNetworkAddressString handle that properly already?